### ISSUE TITLE
Golden Path

### ISSUE THREAD

**Issue by alan-andrade** [pxpgraphics/objective-c-style-guide#1]
For the golden path section https://github.com/pxpgraphics/objective-c-style-guide#golden-path

I want to kick off a discussion for this standard because I think that readability and cognitive ease might be a better way of rationalizing code.

When I read an `if` clause, I'm interested in knowing as fast as I can what the condition is. This will help me out to figure out **why** the conditional exists and also how the branching works.

I think is fairly normal to have a conditional which do a negation just for the sake of following the golden path. The example on this guide does. I don't like this because I takes longer to read the code just because the amount of operations you have to do in your head is greater.

I also think that this practice might be just trying to avoid `if`s nesting which I agree is something bad.

**Comment by pxpgraphics** [comment]
Nice! I agree with a lot of what you've said. :] I definitely think that this is a difficult practice to follow consistently. Although, I strongly agree with your last statement about nesting conditionals, I also would like to prevent duplicate `return`s within a single method, if possible.

It's easier to follow the flow of the code when the there aren't 2, 3, or more `return` statements scattered throughout.

**Comment by alan-andrade** [comment]
That's true, many return statements are a sign of too much branching in a method and that's "no bueno".
